### DISCUSSION TITLE
Current Backup/Recovery Recommended Practices

### DISCUSSION THREAD

**Discussion by adampankow** [Security-Onion-Solutions/securityonion#3444] [Unsupported Versions] [answered]
Is the prevailing guidance still to just redeploy if a significant enough of a failure happens, configuration, hardware, or otherwise? Do people use snapshotting filesystems to revert? Virtualized nodes, each on their own dedicated host? I saw the bit about potentially virtualizing the manager node, as well as talk of doing so with a sensor (forward?) node, but would think if a bad configuration change hit all three nodes in the stack then that would still leave a search node not easily able to be reverted. If all three had a dedicated host they were virtualized on then I would think it would just be a question of how much performance is lost to a hypervisor, plus some added complexity in the setup.

It didn't appear to me that setup using a configuration file was supported in SO2, if I had properly scanned the documentation. Being able to easily revert back from a tough to recover/debug configuration change seems useful, versus having to redeploy and run through the setup manually again.

What practices are folks using to make disaster recovery more easy and efficient?

Thanks for all/any insights!

**Comment by weslambert** [comment, marked as answer]
We allow for traditional Elasticsearch clustering now, which should help in some regard from a resiliency perspective.  You could also utilize snapshotting via Curator to ship snapshots off to cold storage.  Configuration backups are also generated automatically in `/nsm/backup`.

**Reply by adampankow** [reply]
Hey @weslambert, thank you for the reply. I'm most concerned about the configuration and being able to rapidly redeploy it. The data captured is obviously more ephemeral in nature, so we're not too concerned about it. That all said, what does the recovery process with a TAR from `/nsm/backup` typically look like? I checked the docs, but didn't seem to find any information indicating what a backup restoration would consist of.

**Reply by darth-drew** [reply]
@adampankow You may want to keep an eye on #3731
